src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -74,5 +76,124 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapHighWatermark(uint64_t & cu
 #endif
 }
 
+CHIP_ERROR DiagnosticDataProviderImpl::GetRebootCount(uint16_t & rebootCount)
+{
+    uint32_t count = 0;
+    CHIP_ERROR err = ConfigurationMgr().GetRebootCount(count);
+
+    if (err == CHIP_NO_ERROR)
+    {
+        VerifyOrReturnError(count <= UINT16_MAX, CHIP_ERROR_INVALID_INTEGER_VALUE);
+        rebootCount = static_cast<uint16_t>(count);
+    }
+
+    return err;
+}
+
+CHIP_ERROR DiagnosticDataProviderImpl::GetBootReason(uint8_t & bootReason)
+{
+#if CONFIG_HWINFO
+    uint32_t reason = 0;
+
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    if (hwinfo_get_reset_cause(&reason) != 0)
+    {
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    if (reason & (RESET_POR | RESET_PIN))
+    {
+        bootReason = EMBER_ZCL_BOOT_REASON_TYPE_POWER_ON_REBOOT;
+    }
+    else if (reason & RESET_WATCHDOG)
+    {
+        bootReason = EMBER_ZCL_BOOT_REASON_TYPE_SOFTWARE_WATCHDOG_RESET;
```

Comment:
These are problematic since it's a dependency between paltform code and generated code. It would be best if these constants lived in DiagnosticsDataProvider.h and matched the ZCL value. That could be checked where the DiagnosticsDataProvider is used in the cluster, so that there is a static assert for first and last value.

This would avoid the dependency chain being that platform depends on IM/DM and vice-versa, and rather it would just be cluster server that depends on both.

Reply:
@tcarmelveilleux thanks for accurate suggestion, I tried to address it.